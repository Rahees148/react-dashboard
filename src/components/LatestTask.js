import styled from 'styled-components';
//Context
import {useContext} from 'react';
import {TaskContext} from '../context/TaskContext';

const Card = styled.div`
  width:100%;
  padding:24px;
  background-color:#ffffff;
  box-shadow: 0px 3px 6px #0000000A;
  border-radius: 12px;
`;
const Title = styled.h3`
  color: #537178;
  font-size:20px;
  line-height:24px;
  font-weight:500;
  margin:0px 0 24px;
`;
const Ul = styled.ul`
  Padding-left:24px;
`;
const Li = styled.li`
  color: #8F9EA2;
  font-size:14px;
  line-height:26px;
  text-decoration: ${props => props.completed === true? 'line-through' : 'none'};
`;

function LatestTask(){
    const context = useContext(TaskContext);
    const taskList = context.tasks;
    return (
        <Card>
          <Title>Latest Created Tasks</Title>
          <Ul>
          {
              taskList.slice(0,3).map(item => (
                <Li key={item.id} completed={item.completed}> {item.name.length > 30 ? `${item.name.substring(0, 30)}...`:item.name} </Li>
              ))
            }
          </Ul>
        </Card>
    )
}

export default LatestTask;