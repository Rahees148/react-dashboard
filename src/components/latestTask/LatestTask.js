
import {Style} from './LatestTask.style'

//Context
import {useContext} from 'react';
import {TaskContext} from '../../context/TaskContext';


function LatestTask(){
    const context = useContext(TaskContext);
    const taskList = context.tasks;
    return (
        <Style.Card>
          <Style.Title>Latest Created Tasks</Style.Title>
          <Style.Ul>
          {
              taskList.slice(0,3).map(item => (
                <Style.Li key={item.id} completed={item.completed}> 
                  {item.name.length > 30 ? `${item.name.substring(0, 30)}...`:item.name} 
                </Style.Li>
              ))
            }
          </Style.Ul>
        </Style.Card>
    )
}

export default LatestTask;