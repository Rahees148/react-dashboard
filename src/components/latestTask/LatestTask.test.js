
import renderer from 'react-test-renderer';
import LatestTask from './LatestTask';


it('renders correctly', () => {
  const tree = renderer
    .create(<LatestTask />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});