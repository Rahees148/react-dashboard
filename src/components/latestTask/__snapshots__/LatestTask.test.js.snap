// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ iUEVFm"
>
  <h3
    className="sc-gsTCUz XHuqs"
  >
    Latest Created Tasks
  </h3>
  <ul
    className="sc-dlfnbm liKKHL"
  />
</div>
`;
